public/app/features/query/components/QueryEditorRow.tsx

```diff
@@ -415,7 +415,13 @@ export class QueryEditorRow<TQuery extends DataQuery> extends PureComponent<Prop
     const colour = type === 'warning' ? 'orange' : 'blue';
     const iconName = type === 'warning' ? 'exclamation-triangle' : 'file-landscape-alt';
 
-    const serializedWarnings = uniqueWarnings.map((warning) => warning.text).join('\n');
+    let serializedWarnings;
+    if (uniqueWarnings.length > 1) {
+      let textWarnings = uniqueWarnings.map((warning) => <li>{warning.text}</li>);
+      serializedWarnings = <ul>{textWarnings}</ul>;
```

Comment:
We shouldn't use `ul` or `li` like this, we should use proper React elements/components

Reply:
Sorry I'm not familiar with this stuff, can you suggest a suitable React component please?

Reply:
Aah I'm so sorry thats my bad. Maybe something like this would be more React-like.
```ts
const listItems = uniqueWarnings.map((warning) => warning.text);

    return (
      <Badge
        key={key}
        color={colour}
        icon={iconName}
        text={
          <>
            {uniqueWarnings.length} {pluralize(type, uniqueWarnings.length)}
          </>
        }
        tooltip={<List items={listItems} renderItem={(item) => <>{item}</>} />}
      />
    );
```

Reply:
Thank you for the review and helpful advice! Revised accordingly, now the CI failures seem to be unrelated to the changes in this PR, seems like other PRs have the same issues too.